### ISSUE TITLE
Accessibility: Link text ambiguous [Featured Category]

### ISSUE THREAD

**Issue by vikrampm1** [woothemes/woocommerce#52136]
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

### Issue Description:

This link does not include meaningful text in either the visible text, an aria-label, or hidden screen reader text. It is not clear to screen reader users that the link goes to a category archive page instead of the main shop. If multiple of these blocks are added on the same page, there's no way to differentiate them.

![Image](https://github.com/user-attachments/assets/03590d86-f2a6-4905-9683-1c337cf84d32)

### Component or Page: 

Component: Featured Category Block     

### Name affected element:

Shop Now link

### Relevant code snippets:

`<div class="wp-block-button"><a class="wp-block-button__link wp-element-button" href="https://woo.equalizedigital.com/?product_cat=baby">Shop now</a></div>`

### Recommended fix:

Dynamically get the title of the category that is being linked and add it to the Shop Now link to provide additional context on the link.
 
 There are three possible fixes for this link:
 A) Change the visible link text to something meaningful when heard outside the surrounding context. ""Shop Category Name Now""
 B) Add an aria-label to make the link meaningful. aria-label=""Shop Now: Category Name""
 C) Add hidden screen reader text along with the visible link anchor to add context to the anchor. ""Shop now<span class=""sr-only"">: Category Name</span>""

Ensure that you keep the visible text at the beginning of the aria-label so it works for voice control.

**Comment by Manussakis** [comment]
The text of the call to action button is editable so that editors can change the default "Shop now" text.
- https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-blocks/assets/js/blocks/featured-items/call-to-action.tsx#L33
 
Any fix that includes the category title as part of the button label, whether hidden or not, can conflict with the editor's text.

In my opinion, this issue can be closed since WooCommerce already provides editors with a tool to make the content accessible.

cc: @vikrampm1

**Comment by vikrampm1** [comment]
@amberhinds can you review the comment above and let us know if we can proceed and close this issue along with https://github.com/woocommerce/woocommerce/issues/52138?

**Comment by amberhinds** [comment]
Can the default be changed? The issue with this is that the default is not accessible and many people won't change it.

Can you think of an instance when it would not be correct to have an aria-label that includes the category title give that's where the link goes? (I don't think that users can change the link URL, but let me know if that's wrong.)

The best solution would be to have an aria-label that is "button visible text": "Category Name". Then, no matter what they set the visible text to, it would include that text at the start (passing 2.5.3 Label in Name) and have the clarification of the link's purpose.  Is that possible?

**Comment by Manussakis** [comment]
Hey @amberhinds 

> (I don't think that users can change the link URL, but let me know if that's wrong.)

Yes, users can edit the link URL.

![Image](https://github.com/user-attachments/assets/527b75b1-93c4-49a6-8916-c8b8d1692b44)

> Can the default be changed?

It can, but maybe it's not a good idea. I will explain why.

> The best solution would be to have an `aria-label` that is "button visible text": "Category Name". 

Once the `aria-label` attribute is set, the link text added by the editor will be ignored.
To prepend the "Category name" before the accessible name, we would have to add it to the DOM, inside the `<a>` tag and make it only visible for screen readers.
For example:

```html
<a href="#">
  <span class="screen-reader-text">Category Music</span>
  Shop now
</a>
```

However, themes and plugins can filter the link content through the `render_block` filter. Having the `<span>` tag there can likely be a source of regressions.

Even though I agree that editors are unlikely to change the button label, I don't see WooCommerce failing in accessibility with the current solution since the link text can still be edited. 

I’d like to hear the Woo team’s thoughts on this. Is it worth investing in this implementation? If so, I can look into it even further to come up with a less impactful solution.

cc: @vikrampm1

**Comment by amberhinds** [comment]
If screen reader text is added, it needs to be after the visible text, not before.

What about introducing an aria-label field in the block settings?

**Comment by Manussakis** [comment]
@amberhinds

> What about introducing an aria-label field in the block settings?

IMO this is a great idea. I believe I figured out a way to do it without causing regressions: https://github.com/woocommerce/woocommerce/pull/55351

However, if users are unlikely to edit the visible label of the button, the chances are low that they will update the `aria-label` attribute, but at least WooCommerce is providing them with all the tools to improve the store accessibility.

**Comment by amberhinds** [comment]
Yes, then it becomes and education point. Thank you!

**Comment by Manussakis** [comment]
@MajoMaldonado I believe this issue should be marked as "Verified Fixed" after [this comment](https://github.com/woocommerce/woocommerce/pull/55351#issuecomment-2872830033) from @gigitux 

cc: @amberhinds @vikrampm1

**Comment by MajoMaldonado** [comment]
@Manussakis Done.
